src/common/commonutils/CommandUtils.c

```diff
@@ -15,6 +15,44 @@ static long MonotonicTime()
     return -1;
 }
 
+struct mockCommand {
+    const char* expectedCommand;
+    bool matchPrefix;
+    const char* output;
+    int returnCode;
+    struct mockCommand* next;
+};
+
+static struct mockCommand* g_mockCommand;
+
+void AddMockCommand(const char* expectedCommand, bool matchPrefix, const char* output, int returnCode)
+{
+    struct mockCommand* mock;
+    mock = malloc(sizeof(struct mockCommand));
+    // Handling memory error in a brutal way as it's only for tests, not for production usage.
+    if (mock == NULL)
+    {
+        printf("Memory error!");
```

Comment:
>printf("Memory error!"); [](http://example.com/codeflow?start=8&length=24)

No printf calls in production code please! This code runs in daemon context and such printf or other direct console writes add to the syslog logs and that's a problem we want to avoid.

Please:

1. Do not have mocking or other test specific code in production code. Place such code in the unit-tests.

2. Never use printf or alike from production code, only use the OSConfig logging macros which are well protected against unintended syslog routing and log to console when running as an app.